### PR TITLE
Merges 1.13 back to master

### PR DESCRIPTION
**Pull request by elena-kolevska**

### CODE REVIEW

File: .github/workflows/dapr-perf-components.yml [hunk 1]
```diff
@@ -222,7 +222,7 @@ jobs:
         id: setup-go
         uses: actions/setup-go@v5
         with:
-          go-version-file: 'go.mod'
+          go-version: '1.21.9'
```
Comment (artursouza) [inline .github/workflows/dapr-perf-components.yml:225]: Maybe revert these changes?

File: go.sum [hunk 2]
```diff
@@ -437,8 +437,8 @@ github.com/dancannon/gorethink v4.0.0+incompatible h1:KFV7Gha3AuqT+gr0B/eKvGhbjm
 github.com/dancannon/gorethink v4.0.0+incompatible/go.mod h1:BLvkat9KmZc1efyYwhz3WnybhRZtgF1K929FD8z1avU=
 github.com/danieljoos/wincred v1.1.2 h1:QLdCxFs1/Yl4zduvBdcHB8goaYk9RARS2SgLLRuAyr0=
 github.com/danieljoos/wincred v1.1.2/go.mod h1:GijpziifJoIBfYh+S7BbkdUTU4LfM+QnGqR5Vl2tAx0=
-github.com/dapr/components-contrib v1.13.0-rc.2.0.20240503231149-1f46231d875c h1:vzu6TjW2XYZAQY+g9fHmTzOGYNB1lPvRAun7YLV73Nk=
-github.com/dapr/components-contrib v1.13.0-rc.2.0.20240503231149-1f46231d875c/go.mod h1:8/+3UcZLcNytOKLXPpseDT3gB0Mo4ryoMaiud+9u60k=
+github.com/dapr/components-contrib v1.13.1 h1:fNiyi9IEaKmdpEuTjebI8FktR0WUQ/4jq4SnszIqp3Y=
```
Comment (artursouza) [inline go.sum:440]: Revert?

File: go.mod [hunk 3]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/argoproj/argo-rollouts v1.4.1
 	github.com/cenkalti/backoff/v4 v4.2.1
 	github.com/cloudevents/sdk-go/v2 v2.14.0
-	github.com/dapr/components-contrib v1.13.0-rc.2.0.20240503231149-1f46231d875c
+	github.com/dapr/components-contrib v1.13.1
```
Comment (artursouza) [inline go.mod:13]: Revert?

**Review by artursouza — CHANGES_REQUESTED** [review]

File: .github/workflows/dapr-perf-components.yml (hunk 1 above)
Comment (elena-kolevska) [reply .github/workflows/dapr-perf-components.yml]: I'm sorry, I had corrected these, but pushed to the wrong branch. Fixed now.

File: go.mod (hunk 3 above)
Comment (elena-kolevska) [reply go.mod]: The problem was the azure sdk version, but it's already been fixed in components-contrib master, so we can point to the latest commit, for now, until we tag a version, if that's ok.

**Comment by elena-kolevska** [comment]
Moved the membership fixes to https://github.com/dapr/dapr/pull/7817 so they're expected to fail here.

File: go.mod [hunk 4]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/argoproj/argo-rollouts v1.4.1
 	github.com/cenkalti/backoff/v4 v4.2.1
 	github.com/cloudevents/sdk-go/v2 v2.14.0
-	github.com/dapr/components-contrib v1.13.0-rc.2.0.20240503231149-1f46231d875c
+	github.com/dapr/components-contrib v0.0.0-20240618132727-a385743e35c4
```
Comment (artursouza) [inline go.mod:13]: Remove contrib change.

**Comment by artursouza** [comment]
Since this is not an actual merge, as changes were squashed already, we are OK to do a squash merge.
